### DISCUSSION TITLE
dispatch_sync vs. notifyNow

### DISCUSSION THREAD

**Discussion by CodeTechNL** [laravel/framework#46967] [Q&A]
Hi!

its something very small, but when I was using job dispatching and notifications, it caught my eye that there are 2 methods which do the same but named differently. I'm talking about the methods `$user->notifyNow()` for notifications and `dispatch_sync()` for jobs. Isn't it more consistent when `notifyNow()` will be changed to `notifySync()`?

**Comment by travisfont** [comment]
To improve performance, you can queue notifications instead of sending them immediately. To do this, implement the ShouldQueue interface and use the Queueable trait in your notification class.

```php
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Notifications\Notification;
use Illuminate\Bus\Queueable;
```

Updating the class definition to **implement** the ShouldQueue interface and use the Queueable trait:

```php
class MyNotification extends Notification implements ShouldQueue
{
    use Queueable;
    // ...
}
```

This is the most important part as it makes the behavior of the class now asynchronous (rather than synchronously) as notifications will be queued and processed in the background (which is a design to improve the overall performance of an application)

**Reply by CodeTechNL** [reply]
That is not what this topic is about, it's about code consistency, not about performance.

**Comment by y8a5y** [comment]
Hi

Laravel uses `dispatchSync` since `dispatchNow` was removed some version ago, and the renaming was needed to make a clear distinction between the two logic (`dispatchSync` uses a `sync` queue whereas `dispatchNow` was basically directly calling `handle`).

I may be wrong but `notifyNow` does not use the `sync` queue, so that's enough of a difference to not need a renaming imo since the underlying logic is not the same.
